test/extended/images/mysql_replica.go

```diff
@@ -0,0 +1,266 @@
+}
+
+// pollForPods will call getPods in a loop until the required number of pods
+// is returned
+func pollForPods(oc *exutil.CLI, label labels.Selector, count int, timeout time.Duration) ([]string, error) {
+	var pods []string
+	err := wait.Poll(1*time.Second, timeout, func() (bool, error) {
+		p, e := getPods(oc, label)
+		if e != nil {
+			return true, e
+		}
+		if len(p) < count {
+			return false, nil
+		}
+		pods = p
+		return true, nil
+	})
+	return pods, err
+}
+
+type mysql struct {
+	oc      *exutil.CLI
+	PodName string
+	Env     map[string]string
+}
+
+func NewMysql(oc *exutil.CLI, podName, service string) (*mysql, error) {
+	mysql := &mysql{
+		oc:      oc,
+		PodName: podName,
+	}
+
+	pod, err := oc.KubeREST().Pods(oc.Namespace()).Get(podName)
+	if err != nil {
+		return nil, err
+	}
+	mysql.Env = make(map[string]string, len(pod.Spec.Containers[0].Env))
+	for _, env := range pod.Spec.Containers[0].Env {
+		mysql.Env[env.Name] = env.Value
+	}
+	return mysql, nil
+}
+
+// WaitForMySQL pings the MySQL server
+func (m *mysql) WaitForMySQL() (bool, error) {
+	out, err := m.oc.ExecWithoutExit(m.PodName, "mysqladmin -h "+"127.0.0.1"+" -u"+"root"+" ping").Output()
+	if err != nil {
+		switch err.(type) {
+		case *exutil.ExitError:
+			return false, nil
+		default:
+			return false, err
+		}
+	}
+	return out == "mysqld is alive", nil
+}
+
+// WaitUntilUp continuously waits for the server to respond to pings, up until timeout.
+func (m *mysql) WaitUntilUp(timeout time.Duration) error {
+	return wait.Poll(2*time.Second, timeout, m.WaitForMySQL)
+}
+
+// TODO: Generalize, move to exutil
+func (m *mysql) QueryAsRoot(sql string) (string, error) {
+	return m.oc.ExecWithoutExit(m.PodName, "mysql -h "+"127.0.0.1"+" -uroot -e \""+sql+"\" "+m.Env["MYSQL_DATABASE"]).Verbose().Output()
+}
+
+// TODO: Generalize, move to exutil
+func (m *mysql) QueryAsUser(sql string) (string, error) {
+	return m.oc.ExecWithoutExit(m.PodName, "mysql -h "+"127.0.0.1"+" -u"+m.Env["MYSQL_USER"]+" -p"+m.Env["MYSQL_PASSWORD"]+" -e \""+sql+"\" "+m.Env["MYSQL_DATABASE"]).Verbose().Output()
+}
+
+// TODO: Generalize, move to exutil
+func (m *mysql) WaitForMySQLOutput(timeout time.Duration, root bool, sql, resultSubstr string) error {
+	return wait.Poll(5*time.Second, timeout, func() (bool, error) {
+		out, err := m.QueryAsRoot(sql)
+		if err != nil {
+			return false, err
+		}
+		if strings.Contains(out, resultSubstr) {
+			return true, nil
+		}
+		return false, nil
+	})
+}
+
+var _ = g.Describe("images: MySQL replication template", func() {
+	defer g.GinkgoRecover()
+	var (
+		volumeTmpDir string
+		templatePath = "https://raw.githubusercontent.com/openshift/mysql/master/5.5/examples/replica/mysql_replica.json"
+		oc           = exutil.NewCLI("mysql-replication", exutil.KubeConfigPath())
+		user         = "user"
+		userPass     = "userpass"
+		rootPass     = "rootpass"
+		database     = "userdb"
+	)
+
+	g.Describe("MySQL replication", func() {
+		g.JustBeforeEach(func() {
```

Comment:
Yes, that's why I created the CreatePersistentVolume function. BTW, we don't use nfs anymore. But yeah, I guess I'll move that as well.